test/transform_test.cc

```diff
@@ -25,30 +25,48 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include "iceberg/transform/transform_function.h"
+#include "iceberg/type.h"
 #include "iceberg/util/formatter.h"
 
 namespace iceberg {
 
 TEST(TransformTest, TransformFunction) {
-  class TestTransformFunction : public TransformFunction {
-   public:
-    TestTransformFunction() : TransformFunction(TransformType::kUnknown) {}
-    expected<ArrowArray, Error> Transform(const ArrowArray& input) override {
-      return unexpected(
-          Error{.kind = ErrorKind::kNotSupported, .message = "test transform function"});
-    }
-  };
-
-  TestTransformFunction transform;
-  EXPECT_EQ(TransformType::kUnknown, transform.transform_type());
-  EXPECT_EQ("unknown", transform.ToString());
-  EXPECT_EQ("unknown", std::format("{}", transform));
+  IdentityTransform transform{std::make_shared<StringType>()};
```

Comment:
not for this PR but we should consider singleton type instances for common types (though maybe sharing a single refcount is actually a net negative)